spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/CreateChangelogViewProcedure.java

```diff
@@ -146,10 +147,16 @@ public InternalRow[] call(InternalRow args) {
     Dataset<Row> df = loadRows(changelogTableIdent, options(input));
 
     boolean netChanges = input.asBoolean(NET_CHANGES, false);
+    String[] identifierColumns = identifierColumns(input, tableIdent);
+    Preconditions.checkArgument(
+        identifierColumns.length > 0
+            || Arrays.stream(df.schema().fields())
+                .allMatch(field -> OrderUtils.isOrderable(field.dataType())),
+        "Identifier field is required if table contains unorderable columns");
```

Comment:
I think it would be informative if we can log the column name(and type) that is not order-able

Reply:
I agree it would be informative to know which column, but I feel the desired result would be add identifier column to the table or procedure instead to uniquely identify the row instead of drop the column/update the type.

Reply:
I have to agree with @karuppayya that, the error message now is more vague than before 
```
The `sortorder` does not support ordering on type "MAP<STRING, STRING>"
```
I got that we want user to specify the identifier columns, but it's still valuable to provide more error details, so that users can make the choice by themselves.

Reply:
thank you @flyrain and @karuppayya and that's a good point to judge based on existing error message , added unorderable column name as suggested.